src/graderPublic/java/h09/sequence/operation/FlatteningTransformingSequenceBasicTest.java

```diff
@@ -50,8 +50,8 @@ void testIteratorBasic() {
         Sequence<Integer> limitedSeq = new LimitSequence<>((Sequence<Integer>) seq, 6);
         Iterator<Integer> it = limitedSeq.iterator();
         for (int i = 0; i < 6; i++) {
-            Assertions.assertTrue(it.hasNext(), "FlatteningTransformingSequence should have 6 elements");
-            Assertions.assertEquals(i + 1, it.next(), "FlatteningTransformingSequence should have elements 1, 2, 3, 4, 5, 6");
+            Assertions.assertTrue(it.hasNext(), i == 1 ? "Resulting sequence should have 6 elements! Has: 1 element" : "Resulting sequence should have 6 elements! Has: " + i + " elements");
```

Comment:
There is still an off-by-one error in the special case. I would limit the ternary operator to only the last part of the string to avoid duplicating the first part.

Reply:
Oh, I missed that it's now `i == 0`. Still, its better to put the ternary operator at the very end of the string

Reply:
Has been adjusted.